app/lib/features/home/views/home_body.dart

```diff
@@ -0,0 +1,528 @@
+                        // adjust to navbar as primary to smaller views.
+                        Breakpoints.smallAndUp: SlotLayout.from(
+                          key: Keys.mainNav,
+                          inAnimation: AdaptiveScaffold.bottomToTop,
+                          outAnimation: AdaptiveScaffold.topToBottom,
+                          builder: (BuildContext ctx) => BottomNavigationBar(
+                            showSelectedLabels: false,
+                            showUnselectedLabels: false,
+                            currentIndex: widget.navigationShell.currentIndex,
+                            onTap: (index) {
+                              widget.navigationShell.goBranch(
+                                index,
+                                initialLocation: index ==
+                                    widget.navigationShell.currentIndex,
+                              );
+                            },
+                            items: bottomBarNav,
+                            type: BottomNavigationBarType.fixed,
+                          ),
+                        ),
+                      },
+                    )
+                  : null,
+            ),
+          ),
+        ),
+      ),
+    );
+  }
+
+  void onStateChange(VerificationState? prev, VerificationState next) {
+    if (prev == next) return; // avoid too frequent callbacks
+    switch (next.stage) {
+      case 'verification.init':
+        onVerificationInit();
+        break;
+      case 'request.created':
+        onRequestCreated(next.event!);
+        break;
+      case 'request.requested':
+        onRequestRequested(next.event!);
+        break;
+      case 'request.ready':
+        onRequestReady(next.event!, next.isVerifier);
+        break;
+      case 'request.transitioned':
+        onRequestTransitioned(next.event!);
+        break;
+      case 'request.done':
+        onRequestDone(next.event!, next.isVerifier);
+        break;
+      case 'request.cancelled':
+        onRequestCancelled(next.event!, next.isVerifier, next.flowId);
+        break;
+      case 'verification.request':
+        onVerificationRequest(next.event!);
+        break;
+      case 'verification.ready':
+        onVerificationReady(next.event!);
+        break;
+      case 'sas.started':
+        onSasStarted(next.event!, next.isVerifier);
+        break;
+      case 'sas.accepted':
+        onSasAccepted(next.event!, next.isVerifier);
+        break;
+      case 'sas.cancelled':
+        onSasCancelled(next.event!, next.isVerifier, next.flowId);
+        break;
+      case 'sas.keys_exchanged':
+        onSasKeysExchanged(next.event!, next.isVerifier, next.keysExchanged);
+        break;
+      case 'sas.confirmed':
+        onSasConfirmed(next.event!);
+        break;
+      case 'sas.done':
+        onSasDone(next.event!);
+        break;
+    }
+  }
+
+  void onVerificationInit() {
```

Comment:
none of these is technically necessary to be on the shell-widget, isn't it? How about moving them into a separate widget for the cross-singing only then just to clean things clean and separated...

Reply:
I removed `event_emitter` and made `verificationStateProvider` according to your recommendation.
`ref.listen(verificationStateProvider, onStateChange)` should be called in `build()` of `HomeBody`, before that provider can work properly.
Do you want cross signing widget that has no elements and shows only popup dialog for verification?